### PR TITLE
Use Pa11y 7, require Node 18, fix issues

### PR DESCRIPTION
**Pull request by danyalaytekin**
This PR updates the service to use `pa11y@7`, updates the minimum version of Node.js required to `18`, and makes several other changes to modernise the project.

### Changes

- Dependencies
    - Raise minimum Node.js requirement to `18` from `12`
    - Use `pa11y@7` in place of `6`
    - Use `hapi@21` in place of `20`
    - Use `actions@setup-node@4` in place of `3`
    - Remove `request`
- Code
    - Restore CI build's Ubuntu environment to `ubuntu-latest` from `ubuntu-20.04`
    - Allow ES2020 syntax, and use it in some places to reduce repetition & tighten signatures
    - Replace remaining use of `make` with `npm` scripts
    - Use `util.promisify` to flatten some async code
    - Replace use of `request` with `fetch` throughout
    - Rename some ambiguous use of `env` to `mode` for clarity
    - Remove some comments where they repeated adjacent code names
    - Rewrite several tests to `async` syntax, replacing `done` syntax
    - Tidy up test setup
- Documentation
    - Document an option to use `nektos/act` to test locally

### Changes during review process

- [x] https://github.com/pa11y/pa11y-webservice/pull/149/commits/d8d977ef1c339f8224bc3d891c686c8f81c7ea5c Add link to [Docker Desktop](https://www.docker.com/products/docker-desktop/)
- [x] https://github.com/pa11y/pa11y-webservice/pull/149/commits/d24182df76c115537d36aa48dc133237bb01978a Add link to [nektosact.com](https://nektosact.com) and mention other ways to install `act`
- [x] https://github.com/pa11y/pa11y-webservice/pull/149/commits/0f9b6e01b53f60a2a5db33d86a4d5aa23a7766b1 Be clearer that `act` is for locally testing GitHub Actions
- [x] https://github.com/pa11y/pa11y-webservice/pull/149/commits/d857fb18253eb889dfa466ed0bdda442fca3632a Mention installation options across both Docker Desktop and `act`; use shortlinks for consistency

Thanks for the review and documentation fixes @hollsk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,34 +134,65 @@ There are many ways to contribute to Pa11y Webservice, we cover these in the [co
...
 1. ```sh
-   make test-integration    # Run the integration tests
-   make test                # Run both the integration tests and the unit tests mentioned above
+   npm run test:integration   # Run the integration tests
+   npm test                   # Run both the integration tests and the unit tests mentioned above
+   ```
+
+### Testing the GitHub Actions workflow `test.yml`
+
+1. Install Docker Desktop
```
Comment (hollsk) [inline README.md:159]:
```suggestion
1. Install [Docker Desktop](https://www.docker.com/products/docker-desktop/)
```

File: README.md [hunk 2]
```diff
@@ -136,34 +134,65 @@ There are many ways to contribute to Pa11y Webservice, we cover these in the [co
...
+1. Install Nektos Act. For example, with Homebrew:
```
Comment (hollsk) [inline README.md:160]:
```suggestion
1. Install [Nektos Act](https://nektosact.com/). You can install it directly, or with a software package manager. For example, with Homebrew:
```

File: README.md [hunk 3]
```diff
@@ -136,34 +134,65 @@ There are many ways to contribute to Pa11y Webservice, we cover these in the [co
...
    The `&` places the service into the background. An alternative approach is to run `NODE_ENV=test npm start`, suspend the process with `CTRL+z`, and finally run `bg` to place it into the background.
+
 1. ```sh
-   make test-integration    # Run the integration tests
-   make test                # Run both the integration tests and the unit tests mentioned above
+   npm run test:integration   # Run the integration tests
+   npm test                   # Run both the integration tests and the unit tests mentioned above
+   ```
+
+### Testing the GitHub Actions workflow `test.yml`
```
Comment (hollsk) [inline README.md:157]:
```suggestion
### Locally testing the GitHub Actions workflow `test.yml`
```

I've got this right, haven't I? The aim here is to give people a way to test it locally? Reject my suggestion if not, please 😅

**Review by hollsk — APPROVED** [review]
Superb, thank you so much @danyalaytekin 🙏 

There's a few lint warnings in `model/task.js`. All of them are about complexity though, and i think it's probably more important at this point to just get this merged and worry about decomplexifying those blocks later. 

Aside from that, i just added suggestions with direct links to Docker and Act.

File: README.md (hunk 1 above)
Comment (danyalaytekin) [reply README.md:159]: Thanks 👍 I'll apply this and then maybe move it to the bottom, perhaps along with any other `inline` links in here if there any, and mention it's possible with Homebrew as well, as with `act` below.

File: README.md (hunk 3 above)
Comment (danyalaytekin) [reply README.md:157]: That's clearer 🙂

**Comment by danyalaytekin** [comment]
Thanks @hollsk, I've committed those and added one more (list is at the bottom of the PR description).  I don't think the linters' warnings were introduced by this PR, including the nesting one since the previous version had nesting 3 as well, unless the rule doesn't apply to inline functions, but I agree about placating them at some point for the various codebases they're affecting.

**Comment by hollsk** [comment]
Oh yes, love opening up a codebase and finding ancient lint errors in it 😆 Looks perfect, thank you!

**Review by hollsk — APPROVED** [review]

**Pull request merged**